sentry-core/src/main/java/io/sentry/core/SendCachedEventFireAndForgetIntegration.java

```diff
@@ -34,25 +36,21 @@ public final void register(@NotNull IHub hub, @NotNull SentryOptions options) {
       return;
     }
 
-    SendFireAndForget sender = factory.create(hub, options);
+    final SendFireAndForget sender = factory.create(hub, options);
 
     try {
-      ExecutorService es = Executors.newSingleThreadExecutor();
-      es.submit(
-          () -> {
-            try {
-              sender.send();
-              options
-                  .getLogger()
-                  .log(SentryLevel.DEBUG, "Finished processing cached files from %s", cachedDir);
-            } catch (Exception e) {
-              options.getLogger().log(SentryLevel.ERROR, "Failed trying to send cached events.", e);
-            }
-          });
       options
-          .getLogger()
-          .log(SentryLevel.DEBUG, "Scheduled sending cached files from %s", cachedDir);
-      es.shutdown();
+          .getExecutorService()
+          .submit(
+              () -> {
+                try {
+                  sender.send();
+                } catch (Exception e) {
+                  options
+                      .getLogger()
+                      .log(SentryLevel.ERROR, "Failed trying to send cached events.", e);
```

Comment:
H: This API is error prone because we take `Object...` for the string format but the `Throwable` argument is the first:

```suggestion
                      .log(SentryLevel.ERROR, e, "Failed trying to send cached events.");
```

Reply:
yes, also noticed that, I already fixed all the wrong args btw, but it'd be nice to fix that

Reply:
not on this PR though